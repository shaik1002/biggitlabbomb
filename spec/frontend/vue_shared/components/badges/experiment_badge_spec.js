import { mount } from '@vue/test-utils';
import ExperimentBadge from '~/vue_shared/components/badges/experiment_badge.vue';

describe('Experiment badge component', () => {
  let wrapper;

  const createWrapper = (props = {}) => {
    wrapper = mount(ExperimentBadge, {
      propsData: { ...props },
    });
  };

  it('renders the badge', () => {
    createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
