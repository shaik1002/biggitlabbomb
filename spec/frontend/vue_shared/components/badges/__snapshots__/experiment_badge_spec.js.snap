// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiment badge component renders the badge 1`] = `
<div>
  <a
    class="badge badge-neutral badge-pill gl-badge gl-cursor-pointer"
    href="#"
    target="_self"
  >
    <span
      class="gl-badge-content"
    >
      Experiment
    </span>
  </a>
  <div
    class="gl-popover"
  >
    <p>
      An experiment is not yet production-ready, but is released for initial testing and feedback during development.
    </p>
    <p
      class="gl-mb-0"
    >
      Experiments:
    </p>
    <ul
      class="gl-pl-4"
    >
      <li>
        Might be unstable or cause data loss.
      </li>
      <li>
        Are not supported and might not be documented.
      </li>
      <li>
        Could be changed or removed at any time.
      </li>
      <li>
        Are subject to the GitLab Testing Agreement.
      </li>
    </ul>
  </div>
</div>
`;
